### PR TITLE
Callouts in DCR - PR 2/5 - Add more callout components 

### PR DESCRIPTION
**Pull request by tkgnm**
# Callouts in DCR

This PR is part of a larger body of work to create a new Callout component in DCR. We are separating this PR into chunks so that it is easier to communicate the intent of each PR and to make it easier for DCR to review. After each PR is merged, we can merge it into the next branch which will incrementally reduce the size of each subsequent PR. 

Important note for DCR reviewers: Please review these PR's in order below! 

PR 1 https://github.com/guardian/dotcom-rendering/pull/6741
PR 2 https://github.com/guardian/dotcom-rendering/pull/6819
PR 3 https://github.com/guardian/dotcom-rendering/pull/6833
PR 4 https://github.com/guardian/dotcom-rendering/pull/6806
PR 5 https://github.com/guardian/dotcom-rendering/pull/6836

## Why?

This PR adds some new visual components that we will need to create the callout component in a subsequent PR.
 These components are not being used anywhere in production yet. 

## What does this change?

1. Adds a 'description' component
2. Adds a 'share callout' component using a new SVG added into source. 
3. Adds a 'terms and conditions' component
4. Adds an 'expired' component - this will be rendered if the callout has expired.

## Screenshots

I have highlighted the separate components in red.

| Description, share, T&C      |
|-------------|
| ![dstc][] |

[dstc]: https://user-images.githubusercontent.com/102960825/208682599-a0373f34-fbca-443a-ac13-0c323d45f2b4.png

| Expired    |
|-------------|
| ![Expired][] |

[Expired]: https://user-images.githubusercontent.com/102960825/208682716-fa8531f1-12e5-4371-a074-7bc2473ae292.png

### CODE REVIEW

File: dotcom-rendering/src/web/components/Callout/CalloutDescription.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { css } from '@emotion/react';
+import { textSans, space, body } from '@guardian/source-foundations';
+import { decidePalette } from '../../lib/decidePalette';
+
+const descriptionStyles = (format: ArticleFormat) =>
+	css`
+		a {
+			color: ${decidePalette(format).text.richLink};
+			border-bottom: 1px solid ${decidePalette(format).text.richLink};
+			text-decoration: none;
```
Comment (tkgnm) [inline dotcom-rendering/src/web/components/Callout/CalloutDescription.tsx:10]: We want the accent colours to change based on the pillar, hence we use decidePalette here

File: dotcom-rendering/src/web/components/Callout/CalloutDescription.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	`;
+
+interface Props {
+	description: string;
+	format: ArticleFormat;
+}
+
+export const CalloutDescription = ({ description, format }: Props) => (
+	<div css={descriptionStyles(format)}>
+		<div dangerouslySetInnerHTML={{ __html: description }}></div>
```
Comment (tkgnm) [inline dotcom-rendering/src/web/components/Callout/CalloutDescription.tsx:28]: The description we get in the JSON contains some html tags, hence we set html here

File: dotcom-rendering/src/web/components/Callout/CalloutShare.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+export const CalloutShare = ({ format }: Props) => {
+	const [isCopied, setIsCopied] = useState(false);
+
+	const onShare = async () => {
+		const url = window.location.href;
+		if ('clipboard' in navigator) {
+			await navigator.clipboard.writeText(url);
+			setIsCopied(true);
```
Comment (tkgnm) [inline dotcom-rendering/src/web/components/Callout/CalloutShare.tsx:36]: This copies a link to the clipboard so readers can share the callout

File: dotcom-rendering/src/web/components/Callout/CalloutShare.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	};
+	if (typeof window === 'undefined' || typeof navigator === 'undefined')
+		return <></>;
+
+	return (
+		<>
+			<div css={shareCalloutStyles}>
+				<div
+					css={css`
+						width: 45px;
```
Comment (tkgnm) [inline dotcom-rendering/src/web/components/Callout/CalloutShare.tsx:48]: This sets the size of the svg

File: dotcom-rendering/src/web/components/Callout/CalloutShare.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+					<Button
+						size="xsmall"
+						priority="subdued"
+						onClick={onShare}
+						css={shareCalloutLinkStyles(format)}
+					>
+						{' '}
+						Please share this callout.
+					</Button>
+					{isCopied && <em> Link copied to clipboard</em>}
```
Comment (tkgnm) [inline dotcom-rendering/src/web/components/Callout/CalloutShare.tsx:64]: Wrapping the link an `em` tag means the screen reader will read this to anyone using a screen reader. Thanks for the tip there @sophie-macmillan

Comment (sophie-macmillan) [reply dotcom-rendering/src/web/components/Callout/CalloutShare.tsx:64]: Not quite! It needs to have the [role="alert"](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/alert_role) attribute. See the [same component in AR](https://github.com/guardian/dotcom-rendering/pull/6800/files#diff-7e7918a3b84020e0100f847598aa1ae7273ec41877d4156b63293921bc35a96eR41) for reference (also worth noting that the styling has been updated so its not italic and is the same as the supporting text on the fields)

Comment (tkgnm) [reply dotcom-rendering/src/web/components/Callout/CalloutShare.tsx:64]: Cool thanks - will implement here too

File: dotcom-rendering/src/web/components/Callout/CalloutDescription.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+const descriptionStyles = (format: ArticleFormat) =>
+	css`
+		a {
+			color: ${decidePalette(format).text.richLink};
+			border-bottom: 1px solid ${decidePalette(format).text.richLink};
+			text-decoration: none;
+		}
+		${textSans.small()}
+		padding-bottom: ${space[4]}px;
+		${body.medium()}
```
Comment (sophie-macmillan) [inline dotcom-rendering/src/web/components/Callout/CalloutDescription.tsx:14]: Do we need this with the textSans on line 12?

File: dotcom-rendering/src/web/components/Callout/CalloutDescription.tsx (hunk 2 above)
Comment (sophie-macmillan) [reply dotcom-rendering/src/web/components/Callout/CalloutDescription.tsx:28]: Looks like the `sanitize-html` library is installed. We should use it to sanitise the description to be safe.

File: dotcom-rendering/src/web/components/Callout/CalloutDescription.tsx (hunk 6 above)
Comment (tkgnm) [reply dotcom-rendering/src/web/components/Callout/CalloutDescription.tsx:14]: we do not - thanks

File: dotcom-rendering/src/web/components/Callout/CalloutDescription.tsx (hunk 2 above)
Comment (tkgnm) [reply dotcom-rendering/src/web/components/Callout/CalloutDescription.tsx]: Thanks - implemented

**Review by abeddow91 — APPROVED** [review]
Nice work!!

**Comment by bryophyta** [comment]
Reviewed this with @Georges-GNM; looks good!

We thought it would be good to add storybook stories for the new components?

**Comment by tkgnm** [comment]
> Reviewed this with @Georges-GNM; looks good!
> 
> 
> 
> We thought it would be good to add storybook stories for the new components?

@bryophyta @Georges-GNM sure! Can do. I thought this may be a good idea but veered away from it because of increasing chromatic spend. Happy to implement however if you think it's a good idea

File: dotcom-rendering/src/web/components/Callout/CalloutComponents.tsx [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+		></div>
+		<div>
+			Please share your story if you are 18 or over, anonymously if you
+			wish. For more information please see our{' '}
+			<a href="https://www.theguardian.com/help/terms-of-service">
+				terms of service
+			</a>{' '}
+			and{' '}
+			<a href="https://www.theguardian.com/help/privacy-policy">
+				privacy policy
```
Comment (Georges-GNM) [inline dotcom-rendering/src/web/components/Callout/CalloutComponents.tsx:48]: Tiny detail but think you're missing a dot at the end of the sentence.

**Comment by bryophyta** [comment]
> > Reviewed this with @Georges-GNM; looks good!
> > We thought it would be good to add storybook stories for the new components?
> 
> @bryophyta @Georges-GNM sure! Can do. I thought this may be a good idea but veered away from it because of increasing chromatic spend. Happy to implement however if you think it's a good idea

Cheers! We are trying to optimise our Chromatic budget, but for what it's worth my feeling is that it makes sense to add stories for components like this. For instance, because lots of components make use of the more general `decidePalette` functionality, there's scope for regressions to be added by accident later by changes which are meant to only affect one of them. Others might have a different perspective, though!

**Comment by tkgnm** [comment]
> > > Reviewed this with @Georges-GNM; looks good!
> > > We thought it would be good to add storybook stories for the new components?
> > 
> > 
> > @bryophyta @Georges-GNM sure! Can do. I thought this may be a good idea but veered away from it because of increasing chromatic spend. Happy to implement however if you think it's a good idea
> 
> Cheers! We are trying to optimise our Chromatic budget, but for what it's worth my feeling is that it makes sense to add stories for components like this. For instance, because lots of components make use of the more general `decidePalette` functionality, there's scope for regressions to be added by accident later by changes which are meant to only affect one of them. Others might have a different perspective, though!

That's a good point. Thanks. I'll implement this now.

File: dotcom-rendering/src/web/components/Callout/CalloutComponents.tsx (hunk 7 above)
Comment (tkgnm) [reply dotcom-rendering/src/web/components/Callout/CalloutComponents.tsx:48]: Well observed. Will fix.

**Comment by tkgnm** [comment]
> > > > Reviewed this with @Georges-GNM; looks good!
> > > > We thought it would be good to add storybook stories for the new components?
> > > 
> > > 
> > > @bryophyta @Georges-GNM sure! Can do. I thought this may be a good idea but veered away from it because of increasing chromatic spend. Happy to implement however if you think it's a good idea
> > 
> > 
> > Cheers! We are trying to optimise our Chromatic budget, but for what it's worth my feeling is that it makes sense to add stories for components like this. For instance, because lots of components make use of the more general `decidePalette` functionality, there's scope for regressions to be added by accident later by changes which are meant to only affect one of them. Others might have a different perspective, though!
> 
> That's a good point. Thanks. I'll implement this now.

Actually - I think it's best to add these to storybook on the 4th PR in this series, which is when I add the parent component that will house these. components. That will capture all of the potential visual regressions in just one story, which should make things simpler and less bloated.  That okay?

**Review by Georges-GNM — APPROVED** [review]

**Pull request merged**